account_invoice_open_sale/models/account_invoice.py

```diff
@@ -0,0 +1,35 @@
+# coding: utf-8
+# © 2016 Opener B.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools.translate import translate, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    has_sale_order = fields.Boolean(compute='_compute_has_sale_order')
+
+    @api.multi
+    def _compute_has_sale_order(self):
+        """ Determines if the button to show sale order will be shown on the
+        invoice """
+        for invoice in self:
```

Comment:
StefanRijnhart 4 hours ago Member

This will perform badly if the set of invoices is large. You can apply a query
select ail.invoice_id FROM account_invoice_line ail JOIN sale_order_line_invoice_rel rel ON ail.id = rel.invoice_line_id WHERE ail_invoice_id IN %s
instead, and then check for each invoice if the id is in the returned set. Take care not to pass an empty list to the SQL as it will result in an invalid query though.